File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        depth = 0
        ans = []

        for ch in s:

            if ch == '(':

                if depth > 0:
                    ans.append(ch)

                depth += 1

            else:

                depth -= 1

                if depth > 0:
                    ans.append(ch)

        return ''.join(ans)

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {

        int depth = 0;

        StringBuilder ans = new StringBuilder();

        for (char ch : s.toCharArray()) {

            if (ch == '(') {

                if (depth > 0) {
                    ans.append(ch);
                }

                depth++;

            } else {

                depth--;

                if (depth > 0) {
                    ans.append(ch);
                }
            }
        }

        return ans.toString();
    }
}